app/src/components/Heading.tsx

```diff
@@ -10,30 +11,60 @@ import {
     IonTitle,
     IonToolbar,
 } from "@ionic/react";
-import { ellipsisHorizontal, ellipsisVertical, personCircle, search } from "ionicons/icons";
+import { ellipsisHorizontal, ellipsisVertical, logOut, personCircle, search } from "ionicons/icons";
+import { useUserContext } from "../utils/useUserContext";
+import { firebaseAuth } from "../services/firebase";
 
 interface HeadingProps {
     title: string | null | undefined;
 }
 
 const Heading: React.FC<HeadingProps> = ({ title }) => {
+    const userContext = useUserContext();
+    const [showAlert, setShowAlert] = useState(false);
+
     return (
-        <>
-            <link href="https://fonts.googleapis.com/css?family=Share Tech Mono" rel="stylesheet"></link>
-            <IonHeader>
-                <IonToolbar color="dark">
-                    <IonButtons slot="start">
-                        <IonBackButton />
-                    </IonButtons>
-                    <IonButtons slot="secondary">
-                        <IonButton>
-                            <IonIcon slot="icon-only" icon={personCircle} />
+        <IonHeader>
+            <IonAlert
+                isOpen={showAlert}
+                onDidDismiss={() => setShowAlert(false)}
+                header={"Logout Confirmation"}
+                message={"Are you sure you want to logout?"}
+                buttons={[
+                    {
+                        text: "Cancel",
+                        role: "cancel",
+                        cssClass: "secondary",
+                        handler: (blah) => {
+                            setShowAlert(false);
+                        },
+                    },
+                    {
+                        text: "Okay",
+                        handler: () => {
+                            firebaseAuth.signOut();
+                        },
+                    },
+                ]}
+            />
+            <IonToolbar color="dark">
+                <div className="w-full flex justify-between items-center">
+                    <IonBackButton />
```

Comment:
The back button isn't showing up for me. I think you need to put the <IonButton> thing around this again for it to work.